### ISSUE TITLE
Set same behaviour downloading from "Open In" as in regular download when device is offline

### ISSUE THREAD

**Issue by jesmrec** [owncloud/ios-app#179]
With no connection:

Clicking on a file to download

![nov-30-2018 12-04-19](https://user-images.githubusercontent.com/14894746/49285586-1dfb1280-f498-11e8-859e-30138dc45896.gif)

"Open In":

![nov-30-2018 12-10-56](https://user-images.githubusercontent.com/14894746/49285898-0ec89480-f499-11e8-93df-b703214e73b0.gif)

- Direct download warn the user he does not have connection
- Open In tries to download forever

Both should have the same behaviour.

**Comment by michaelstingl** [comment]
After performing "Open in…" whilst being offline, progress bar stays forever in broken state. (same in https://github.com/owncloud/ios-app/issues/182)

**Comment by pablocarmu** [comment]
If I understand correctly this would add a new argument (or another kind of parameter/option) to the OCCore's `download` method to download from server or retrieve the file from cache depending if there's a connection or not.

Something like: 

```Swift
// Please don't take attention to the argument name, it's a dummy one
//
// With connection
core.download(..., fromServer: true, ...)

// Without connection
core.download(..., fromServer: false, ...)
```

Is this correct?
if this is correct I totally agree

**Comment by felix-schwarz** [comment]
@pablocarmu I thought of something like this

```swift
core.download(..., options: [ OCDownloadOption.returnImmediatelyIfOfflineOrUnavailable : true ], …)
```

This would behave like this:
- when online: download the item and then return it via the `completionHandler`
- when offline or in maintenance mode and …
    - … a local copy exists: return the local copy via the `completionHandler`
    - … no local copy exists: return an error to the `completionHandler`

**Comment by pablocarmu** [comment]
works for me!

**Comment by felix-schwarz** [comment]
The API is available as of ca11f9421d859724ea7609056fad1342fbc9b18e. Also in there I've changed DisplayViewController to take advantage of the new API to show how it can be used.

**Comment by jesmrec** [comment]
An alert will show the user that he lacks connection, for this first iteration.
